### PR TITLE
Feat(duckdb): add parsing support for timestamp types with unit suffixes

### PR DESCRIPTION
**Pull request by cpcloud**
Add support for parsing duckdb timestamp types with unit suffixes

### CODE REVIEW

File: sqlglot/dialects/duckdb.py [hunk 1]
```diff
@@ -321,6 +325,10 @@ class Generator(generator.Generator):
             exp.DataType.Type.UINT: "UINTEGER",
             exp.DataType.Type.VARBINARY: "BLOB",
             exp.DataType.Type.VARCHAR: "TEXT",
+            exp.DataType.Type.TIMESTAMP_S: "TIMESTAMP_S",
+            exp.DataType.Type.TIMESTAMP_MS: "TIMESTAMP_MS",
+            exp.DataType.Type.TIMESTAMP_US: "TIMESTAMP_US",
```
Comment (cpcloud) [inline sqlglot/dialects/duckdb.py:330]: We could potentially leave this out as parse target and just turn `TIMESTAMP_US` into `TIMESTAMP`, but I wasn't sure exactly how to do that.

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/dialects/duckdb.py (hunk 1 above)
Comment (cpcloud) [reply sqlglot/dialects/duckdb.py]: Done!

**Comment by cpcloud** [comment]
Ah right, thanks!

**Comment by cpcloud** [comment]
Thanks for the quick review!

**Pull request merged**